models/location_helpers.go

```diff
@@ -358,6 +358,21 @@ func RemoveBackupLocation(q *reform.Querier, id string, mode RemoveMode) error {
 		return err
 	}
 
+	artifacts, err := FindArtifacts(q, &ArtifactFilters{LocationID: id})
+	if err != nil {
+		return err
+	}
+
+	if len(artifacts) != 0 && mode == RemoveRestrict {
+		return status.Errorf(codes.FailedPrecondition, "backup location with ID %q has artifacts.", id)
+	}
+
+	for _, a := range artifacts {
+		if err := RemoveArtifact(q, a.ID); err != nil {
+			return err
+		}
+	}
+
 	// @TODO - force delete https://jira.percona.com/browse/PMM-7475
```

Comment:
you basically did this ticket, no?

Reply:
well looks like yes :) it is left to handle `force` flag

Reply:
done